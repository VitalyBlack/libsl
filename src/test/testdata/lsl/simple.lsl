libsl "1.0.0";
library simple version "1.0.0f" language "java" url "https://github.com/vldf/";

import "another/one/file";

include "file.to.include";

types {
   SimpleType(ru.vldf.Type);
   SimpleTypeWithGeneric(ru.vldf.Type<ru.vldf.Generic>);
   EnumLikeType(ru.vldf.Type) {
      VARIANT1: -1;
      VARIANT2: true;
      VARIANT2: 1.0;
      VARIANT3: "String";
   }
}

automaton A {
   state S1;
   state S2;

   shift S1->S2(func);

   var strVar: String;
   var intVar: Int = 1;
   automaton b = new A(state = S1, strVar = "", intVar = 1);

   fun func(arg1: SimpleType): SimpleTypeWithGeneric;
}

automaton B {
   state S1, S3, S7;
   state S2;

   shift S1->S2(func(SimpleTypeWithGeneric));
   shift S2->S1(func(SimpleType));

   var v: Int;

   fun func(arg1: SimpleType): SimpleTypeWithGeneric;
   fun func(arg1: SimpleTypeWithGeneric): SimpleTypeWithGeneric;
}

fun A.extendedFunction() {
   strVar = "";
   b.v = 1;
}
