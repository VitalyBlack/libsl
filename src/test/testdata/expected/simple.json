{
  "metadata": {
    "name": "simple",
    "libraryVersion": "1.0.0f",
    "language": "java",
    "url": "https://github.com/vldf/",
    "lslVersion": {
      "first": {
        "data": 1
      },
      "second": {
        "data": 0
      },
      "third": {
        "data": 0
      }
    }
  },
  "imports": [
    "another/one/file"
  ],
  "includes": [
    "file.to.include"
  ],
  "semanticTypes": [
    {
      "semanticType": "SimpleType",
      "realType": {
        "name": [
          "ru",
          "vldf",
          "Type"
        ]
      }
    },
    {
      "semanticType": "SimpleTypeWithGeneric",
      "realType": {
        "name": [
          "ru",
          "vldf",
          "Type"
        ],
        "generic": [
          "ru",
          "vldf",
          "Generic"
        ]
      }
    },
    {
      "semanticType": "EnumLikeType",
      "realType": {
        "name": [
          "ru",
          "vldf",
          "Type"
        ]
      },
      "entities": [
        {
          "first": "VARIANT1",
          "second": "-1"
        },
        {
          "first": "VARIANT2",
          "second": "true"
        },
        {
          "first": "VARIANT2",
          "second": "1.0"
        },
        {
          "first": "VARIANT3",
          "second": "\"String\""
        }
      ]
    }
  ],
  "automata": [
    {
      "name": "A",
      "kind": "REAL",
      "states": [
        {
          "name": "S1",
          "kind": "SIMPLE",
          "isSelf": false,
          "isAny": false
        },
        {
          "name": "S2",
          "kind": "SIMPLE",
          "isSelf": false,
          "isAny": false
        }
      ],
      "shifts": [
        {
          "from": {
            "name": "S1",
            "kind": "SIMPLE",
            "isSelf": false,
            "isAny": false
          },
          "to": {
            "name": "S2",
            "kind": "SIMPLE",
            "isSelf": false,
            "isAny": false
          },
          "functions": [
            {
              "name": "func",
              "automatonName": "A",
              "args": [
                {
                  "name": "arg1",
                  "type": {
                    "semanticType": "SimpleType",
                    "realType": {
                      "name": [
                        "ru",
                        "vldf",
                        "Type"
                      ]
                    }
                  }
                }
              ],
              "returnType": {
                "semanticType": "SimpleTypeWithGeneric",
                "realType": {
                  "name": [
                    "ru",
                    "vldf",
                    "Type"
                  ],
                  "generic": [
                    "ru",
                    "vldf",
                    "Generic"
                  ]
                }
              },
              "contracts": [],
              "statements": []
            }
          ]
        }
      ],
      "variables": [],
      "localFunctions": [
        {}
      ]
    },
    {
      "name": "B",
      "kind": "REAL",
      "states": [
        {
          "name": "S1",
          "kind": "SIMPLE",
          "isSelf": false,
          "isAny": false
        },
        {
          "name": "S3",
          "kind": "SIMPLE",
          "isSelf": false,
          "isAny": false
        },
        {
          "name": "S7",
          "kind": "SIMPLE",
          "isSelf": false,
          "isAny": false
        },
        {
          "name": "S2",
          "kind": "SIMPLE",
          "isSelf": false,
          "isAny": false
        }
      ],
      "shifts": [
        {
          "from": {
            "name": "S1",
            "kind": "SIMPLE",
            "isSelf": false,
            "isAny": false
          },
          "to": {
            "name": "S2",
            "kind": "SIMPLE",
            "isSelf": false,
            "isAny": false
          },
          "functions": [
            {
              "name": "func",
              "automatonName": "B",
              "args": [
                {
                  "name": "arg1",
                  "type": {
                    "semanticType": "SimpleTypeWithGeneric",
                    "realType": {
                      "name": [
                        "ru",
                        "vldf",
                        "Type"
                      ],
                      "generic": [
                        "ru",
                        "vldf",
                        "Generic"
                      ]
                    }
                  }
                }
              ],
              "returnType": {
                "semanticType": "SimpleTypeWithGeneric",
                "realType": {
                  "name": [
                    "ru",
                    "vldf",
                    "Type"
                  ],
                  "generic": [
                    "ru",
                    "vldf",
                    "Generic"
                  ]
                }
              },
              "contracts": [],
              "statements": []
            }
          ]
        },
        {
          "from": {
            "name": "S2",
            "kind": "SIMPLE",
            "isSelf": false,
            "isAny": false
          },
          "to": {
            "name": "S1",
            "kind": "SIMPLE",
            "isSelf": false,
            "isAny": false
          },
          "functions": [
            {
              "name": "func",
              "automatonName": "B",
              "args": [
                {
                  "name": "arg1",
                  "type": {
                    "semanticType": "SimpleType",
                    "realType": {
                      "name": [
                        "ru",
                        "vldf",
                        "Type"
                      ]
                    }
                  }
                }
              ],
              "returnType": {
                "semanticType": "SimpleTypeWithGeneric",
                "realType": {
                  "name": [
                    "ru",
                    "vldf",
                    "Type"
                  ],
                  "generic": [
                    "ru",
                    "vldf",
                    "Generic"
                  ]
                }
              },
              "contracts": [],
              "statements": []
            }
          ]
        }
      ],
      "variables": [],
      "localFunctions": [
        {},
        {}
      ]
    },
    {
      "name": "Main",
      "kind": "SYNTHETIC",
      "states": [],
      "shifts": [],
      "variables": [],
      "localFunctions": []
    }
  ],
  "extensionFunctions": {
    "A": [
      {
        "name": "extendedFunction",
        "automatonName": "A",
        "args": [],
        "contracts": [],
        "statements": [
          {},
          {}
        ]
      }
    ]
  }
}